test/config.js

```diff
@@ -252,3 +254,47 @@ test("Config override is applied if not specified and already exists at default
 
   t.assert(outputString.includes('"three": 12345678'), "Applies override if exists but not specified")
 })
+
+test('Config edits trigger re-bundle if using watchify', async (t) => {
+  t.plan(2)
+  const configDefault = {
+    resources: {
+      '<root>': {
+        packages: {
+          'two': true
+        }
+      },
+      'two': {
+        packages: {
+          'three': 12345678
+        }
+      }
+    }
+  }
+  const bundler = await createWatchifyBundle({
+    writeAutoConfig: true,
+  })
+
+  const overridePath = './lavamoat/lavamoat-config-override.json'
+  const configPath = './lavamoat/lavamoat-config.json'
+  
+  bundler.emit('file', './lavamoat/lavamoat-config-override.json')
+
+  const configFile = fs.readFileSync(configPath)
+  const configFileString = configFile.toString()
+  
+  const overrideString = JSON.stringify(configDefault)
+  fs.writeFileSync(overridePath, overrideString)
+  
+
+  bundler.on('update', async() => {
```

Comment:
Tests won't wait for this to complete
Use pify or await new promise here for bundle to complete and then run subsequent code

Reply:
Solved, after a while of head banging. In some cases the 'update' event was firing, others not. (even after re-running tests without changing any code)Turned out to be an underlying filesystem watch issue. Set `poll: true` in watchify plugin opts. Does the trick for testing purposes.

Reply:
Weird!